simplemma/language_detector.py

```diff
@@ -10,24 +10,44 @@
 from .dictionary_factory import DictionaryFactory
 from .tokenizer import Tokenizer
 
+
 SPLIT_INPUT = re.compile(r"[^\W\d_]{3,}")
+RELAXED_SPLIT_INPUT = re.compile(r"[\w-]{3,}")
 
 
 class TokenSampler:
-    __slots__ = ["tokenizer"]
+    __slots__ = ["capitalized_threshold", "max_tokens", "tokenizer"]
 
-    def __init__(self, tokenizer: Tokenizer = Tokenizer(SPLIT_INPUT)) -> None:
+    def __init__(
+        self,
+        tokenizer: Tokenizer = Tokenizer(SPLIT_INPUT),
+        max_tokens: int = 1000,
+        capitalized_threshold: float = 0.8,
+    ) -> None:
         self.tokenizer = tokenizer
+        self.max_tokens = max_tokens
+        self.capitalized_threshold = capitalized_threshold
 
     def sample_tokens(self, text: str) -> List[str]:
         """Extract potential words, scramble them, extract the most frequent,
         some of the rest, and return at most 1000 tokens."""
-        # generator expression to split the text
-        counter = Counter(
-            token for token in self.tokenizer.split_text(text) if not token[0].isupper()
-        )
 
-        return [item[0] for item in counter.most_common(1000)]
+        counter = Counter(token for token in self.tokenizer.split_text(text))
+
+        deletions = [token for token in counter if token[0].isupper()]
+        if len(deletions) < self.capitalized_threshold * len(counter):
+            for token in deletions:
+                del counter[token]
+
+        return [item[0] for item in counter.most_common(self.max_tokens)]
+
+
+class RelaxedTokenSampler(TokenSampler):
```

Comment:
With what I proposed above, this could be done just as
```python
class RelaxedTokenSampler(TokenSampler):
    def __init__(self) -> None:
        super().__init__(RELAXED_SPLIT_INPUT, None, None)
```
right?


Or just during constructor

```python
TokenSampler(RELAXED_SPLIT_INPUT, None, None)
```

Instead of creating a bunch of TokenSampler we could have the basic one and just document different possible configurations.
See my comment below.